pandas/core/sorting.py

```diff
@@ -367,9 +370,12 @@ def nargsort(
             mask=mask,
         )
 
-    items = extract_array(items)
+    if isinstance(items, ABCRangeIndex):
+        return items.argsort(ascending=ascending)  # TODO: test coverage with key?
```

Comment:
hmm if you use extract_range on L376 i think you can just fall thru no? (or I guess need L380 to change), anyhow for the future